src/object/cli_obj.c

```diff
@@ -3321,13 +4089,41 @@ obj_list_common(tse_task_t *task, int opc, daos_obj_list_t *args)
 		D_ASSERT(args->dkey_anchor != NULL);
 
 		shard = dc_obj_anchor2shard(args->dkey_anchor);
-		if (obj_auxi->to_leader)
-			shard = obj_grp_leader_get(obj, shard, map_ver);
-		else if (!obj_auxi->is_ec_obj)
+		if (obj_auxi->to_leader) {
+			struct pl_obj_shard *p_shard;
+			int leader;
+
+			leader = obj_grp_leader_get(obj, shard, map_ver);
+			p_shard = obj_get_shard(obj, leader);
+			if (p_shard->po_rebuilding) {
```

Comment:
|| p_shard->po_target == -1 ? can it work when no spare target?

Reply:
hmm, indeed. thanks